pkg/client/client_test.go

```diff
@@ -678,6 +681,54 @@ var _ = Describe("ConmonClient", func() {
 			Expect(response).To(BeNil())
 		})
 	})
+
+	Describe("ServeExecContainer", func() {
+		for _, terminal := range []bool{true, false} {
+			terminal := terminal
+
+			It(testName("should succeed", terminal), func() {
+				tr = newTestRunner()
+				tr.createRuntimeConfigWithProcessArgs(terminal, []string{"/busybox", "sleep", "10"}, nil)
+				sut = tr.configGivenEnv()
+				tr.createContainer(sut, terminal)
+				tr.startContainer(sut)
+
+				result, err := sut.ServeExecContainer(context.Background(), &client.ServeExecContainerConfig{
+					ID:      tr.ctrID,
+					Command: []string{"/busybox", "sh", "-c", "echo -n stdout && echo -n stderr >&2"},
+					Tty:     terminal,
+					Stdout:  true,
+					Stderr:  true,
+				})
+
+				Expect(err).To(Succeed())
+				Expect(result.URL).NotTo(BeEmpty())
+
+				config := &rest.Config{TLSClientConfig: rest.TLSClientConfig{Insecure: true}}
+				executor, err := remotecommand.NewWebSocketExecutor(config, "GET", result.URL)
+				Expect(err).NotTo(HaveOccurred())
+
+				stdout := &bytes.Buffer{}
+				stderr := &bytes.Buffer{}
+
+				streamOptions := remotecommand.StreamOptions{
+					Stdout: stdout,
+					Stderr: stderr,
+					Tty:    terminal,
+				}
+				err = executor.StreamWithContext(context.Background(), streamOptions)
+				Expect(err).NotTo(HaveOccurred())
+
+				if terminal {
+					Expect(stdout.String()).To(Equal("stdoutstderr"))
+					Expect(stderr.Len()).To(BeZero())
+				} else {
+					Expect(stdout.String()).To(Equal("stdout"))
+					Expect(stderr.String()).To(Equal("stderr"))
+				}
+			})
+		}
+	})
```

Comment:
any chance we could have a couple more integration tests?

Reply:
Sure thing!